allennlp/data/dataset_readers/coreference_resolution/conll.py

```diff
@@ -0,0 +1,293 @@
+        self.coref_stacks: DefaultDict[int, List[int]] = collections.defaultdict(list)
+
+    def assert_document_is_finished(self) -> None:
+        if self.sentence_buffer:
+            raise ConfigurationError("Processing was attempted on a document which had "
+                                     "incomplete sentences. Current sentence buffer "
+                                     "contains: {}".format(self.sentence_buffer))
+
+        if not self.sentences or self.num_total_words == 0:
+            raise ConfigurationError("Processing was attempted on a document with no sentences or words.")
+
+        if any(x for x in self.coref_stacks.values()):
+            raise ConfigurationError("Processing was attempted on a document which contains incomplete "
+                                     "(unclosed) spans within the scope of the document. Current ids "
+                                     "and start indices" " of unclosed spans: {}".format(self.coref_stacks))
+
+    def complete_sentence(self) -> None:
+        self.sentences.append(self.sentence_buffer)
+        self.sentence_buffer = []
+
+    def add_word(self, word) -> None:
+        self.sentence_buffer.append(word)
+        self.num_total_words += 1
+
+    def canonicalize_clusters(self) -> List[List[Tuple[int, int]]]:
+        """
+        The CONLL 2012 data includes 2 annotations which have identical
+        spans, but different ids. This checks for spans in all clusters
+        which are equal, and if it finds any, merges the two relevant
+        clusters.
+        """
+        merged_clusters: List[Set[Tuple[int, int]]] = []
+        for cluster in self.clusters.values():
+            cluster_with_overlapping_mention = None
+            for mention in cluster:
+                # Look at clusters we have already processed to
+                # see if they contain a mention in the current
+                # cluster for comparison.
+                for cluster2 in merged_clusters:
+                    if mention in cluster2:
+                        # first cluster in merged clusters
+                        # which contains this mention.
+                        cluster_with_overlapping_mention = cluster2
+                        break
+                # Already encountered overlap - no need to keep looking.
+                if cluster_with_overlapping_mention is not None:
+                    break
+            if cluster_with_overlapping_mention is not None:
+                # Merge cluster we are currently processing into
+                # the cluster in the processed list.
+                cluster_with_overlapping_mention.update(cluster)
+            else:
+                merged_clusters.append(set(cluster))
+        return [list(c) for c in merged_clusters]
+
+
+@DatasetReader.register("coref")
+class ConllCorefReader(DatasetReader):
+    """
+    Reads a single CoNLL-formatted file. This is the same file format as used in the
```

Comment:
A CoNLL 2012 Coref file?

Reply:
Not sure what you're asking/suggesting here.